pkg/trait/logging.go

```diff
@@ -61,7 +61,15 @@ func (l loggingTrait) Configure(e *Environment) (bool, *TraitCondition, error) {
 }
 
 func (l loggingTrait) Apply(e *Environment) error {
-	envvar.SetVal(&e.EnvVars, envVarQuarkusLogLevel, l.Level)
+	if loggingCap, ok := e.CamelCatalog.Runtime.Capabilities["logging"]; ok {
+		if e.ApplicationProperties == nil {
+			e.ApplicationProperties = make(map[string]string)
+		}
+		e.ApplicationProperties[loggingCap.Properties["level"]] = l.Level
```

Comment:
the properties in capabilities may be an array, so in the trait code it just iterates over the key/values. So, there is no need to set individual property keys.

Reply:
If we do an array we'd need to parse the value, ie `test=hello` as key/value with the all the possible problems it would encompass.